Sources/Chat/ProtocolServices/Identity/IdentityService.swift

```diff
@@ -0,0 +1,139 @@
+import Foundation
+
+actor IdentityService {
+
+    private let keyserverURL: URL
+    private let kms: KeyManagementServiceProtocol
+    private let storage: IdentityStorage
+    private let networkService: IdentityNetworking
+    private let iatProvader: IATProvider
+    private let messageFormatter: SIWECacaoFormatting
+
+    init(
+        keyserverURL: URL,
+        kms: KeyManagementServiceProtocol,
+        storage: IdentityStorage,
+        networkService: IdentityNetworking,
+        iatProvader: IATProvider,
+        messageFormatter: SIWECacaoFormatting
+    ) {
+        self.keyserverURL = keyserverURL
+        self.kms = kms
+        self.storage = storage
+        self.networkService = networkService
+        self.iatProvader = iatProvader
+        self.messageFormatter = messageFormatter
+    }
+
+    func registerIdentity(account: Account,
+        onSign: (String) -> CacaoSignature
```

Comment:
as discussed we should request to sign async

Reply:
updated 
created specs PR: https://github.com/WalletConnect/walletconnect-docs/pull/510